### DISCUSSION TITLE
[dict] contains and extract value

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#261] [Q&A] [answered]
How to check if a nanobind::dict object contains a key and extract its value as a specified type like float?
Is there any way like the code below?

nanobind::dict d;
if (d.contains("key"))
{
    float v = d.value<float>("key");
}

**Comment by wjakob** [comment, marked as answer]
instead of `.value`, use `operator[]`. The `contains` function was indeed missing, I added in commit https://github.com/wjakob/nanobind/commit/654dbaa78d9aef12d636e380dec8c50bd2e59b83 just now.

**Reply by huangweiwu** [reply]
The contains function works now!
The operator[] returns an accessor, how to get the result? nanobind::cast<float>?

**Reply by wjakob** [reply]
The accessor supports implicit conversion to `nb::object`, so you can simply assign it to such a variable.

**Comment by gillesdegottex** [comment]
Just as a follow up for anyone stumbling on this. This works:
```
float value = nb::cast<float>(nb::object(kwargs["key"]));
```
both for arguments `key=1` or `key=1.0` (which is handy for code flexibility)
